### PR TITLE
Wires the locale provided by Fuchsia.

### PR DESCRIPTION
**Pull request by filmil**
The FIDL service `fuchsia.intl.PropertyProvider` is a service that
the flutter runner can use to obtain information on system preferred
locales.

This change sends a platform message "setLocale" on the channel
"flutter/localization", based on the values provided by the above
mentioned FIDL service.

Credit: most of this was initially written by @kpozin; I ported it
to out-of-tree flutter engine.

Tested:

1. Compile and publish the unit tests package as shown in
   the script below.
2. In a Fuchsia repository (pointed to by `$FUCHSIA_DIR`), run
   `fx serve`
3. `fx shell run fuchsia-pkg://fuchsia.com/flutter_runner_tests#meta/flutter_runner_tests.cmx`

The script used to update the unit tests.

```bash
#!/bin/bash

set -x

# You will need to ensure that $FLUTTER_ENGINE_DIR and $FUCHSIA_DIR
# are set to appropriate values for your machine.
FLUTTER_ENGINE_DIR="${FLUTTER_ENGINE_DIR:-$HOME/fx/flutter/engine/src}"
readonly OUT_DIR="${FLUTTER_ENGINE_DIR}/out"

(
  cd ${FLUTTER_ENGINE_DIR}
  ./flutter/tools/gn --fuchsia --fuchsia-cpu x64 --unoptimized
  ninja -j 100 -C "${OUT_DIR}/fuchsia_debug_unopt_x64"
  cp "${OUT_DIR}/compile_commands.json" "${FLUTTER_ENGINE_DIR}"

  echo "Publishing the tests package"
  "${FLUTTER_ENGINE_DIR}/fuchsia/sdk/linux/tools/pm" publish \
    -a -r $FUCHSIA_DIR/out/release/amber-files \
    -f "${FLUTTER_ENGINE_DIR}/out/fuchsia_debug_unopt_x64/flutter_runner_tests-0.far"
)
```

### CODE REVIEW

File: shell/platform/fuchsia/flutter/fuchsia_intl.cc [hunk 1]
```diff
@@ -0,0 +1,126 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
```
Comment (chinmaygarde) [inline shell/platform/fuchsia/flutter/fuchsia_intl.cc]: Please use the usual Flutter license block like in the files next to this TU.

File: shell/platform/fuchsia/flutter/fuchsia_intl.cc [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+#include "third_party/icu/source/common/unicode/locid.h"
+#include "third_party/icu/source/common/unicode/strenum.h"
+#include "third_party/icu/source/common/unicode/stringpiece.h"
+#include "third_party/icu/source/common/unicode/uloc.h"
+
+namespace {
+
+using icu::Locale;
+
+// This backfills for the static method icu::Locale::forLanguageTag which is
```
Comment (chinmaygarde) [inline shell/platform/fuchsia/flutter/fuchsia_intl.cc]: We should just update the ICU version ourselves. I did it about a year ago so it is definitely a bit old.

**Comment by chinmaygarde** [comment]
cc @jason-simmons @GaryQian

File: shell/platform/fuchsia/flutter/fuchsia_intl.cc (hunk 1 above)
Comment (filmil) [reply shell/platform/fuchsia/flutter/fuchsia_intl.cc]: Will do. I will just assume that your suggested approach is fine.

File: shell/platform/fuchsia/flutter/fuchsia_intl.cc (hunk 2 above)
Comment (filmil) [reply shell/platform/fuchsia/flutter/fuchsia_intl.cc]: OK. But, let me do so in a separate change then.

Comment (filmil) [reply shell/platform/fuchsia/flutter/fuchsia_intl.cc]: Do you have a preference for the version?  ICU 64 has been released on Oct 3.  64.2 was current before, but it also has a lot of patches that would need to be applied too.

**Comment by cbracken** [comment]
@filmil looks like the ICU upgrade landed but was reverted. I assume you're working on a re-land before picking this up again?

**Comment by filmil** [comment]
@cbracken My understanding is that build cops needed to re-sequence several ICU landings, and will reapply the ICU upgrade that I did.  Once done, they will reapply my change, and I can continue on this.  Will let you know when ready.

**Comment by filmil** [comment]
I had a license file merge to do, and this took a while.  The current commit seems to validate well on my machine, let's see if the pre-commit hooks agree.

**Comment by filmil** [comment]
@chinmaygarde @cbracken PTAL?  This has now passed all checks, whew!

**Comment by filmil** [comment]
Yes, please.

**Comment by filmil** [comment]
Rebased on top of current HEAD.

**Pull request merged**